### DISCUSSION TITLE
Automated Lunar Seeder to Facilitate Testing with RefreshDatabase Trait

### DISCUSSION THREAD

**Discussion by nick-nds** [lunarphp/lunar#1163] [Feature Requests]
So, I've been using the `RefreshDatabase` trait for testing my app, and it clears database before tests. But, here's the catch – I had to manually add those database entries that Lunar installs via seeders. 🌱 And guess what? The `install:lunar` command needs user input, so installing lunar before tests is a no-go. 😫

Here's the idea: What if we had a Lunar seeder specifically for tests, loaded with test data? 🚀 That way, we could easily test our apps without the hassle! Right now, I've set up a simple seeder for my database to work around the issue.

**Comment by netzknecht** [comment]
You can try to call `install:lunar` inside a seeder or test with `Artisan::call()` and skip interaction to process with default values. If you use the lunar hub, it might be required to create an `\Lunar\Hub\Models\Staff` model first and set the `admin` property to `true`.
```
        Artisan::call('lunar:install', [
            '--quiet' => true, 
            '--no-interaction' => true
        ]);
```
The idea to have different seeders with factories for testing and seeders that process some config/template files for setup sounds great.

In my current project I don't use the `install:lunar` command and use custom seeders instead for testing and initial setup. To seed the countries and states I use the artisan command inside a seeder:

```
namespace Database\Seeders;

use Illuminate\Database\Seeder;
use Illuminate\Support\Facades\Artisan;

class AddressSeeder extends Seeder
{
    /**
     * Run the database seeds.
     */
    public function run(): void
    {
        Artisan::call('lunar:import:address-data');
    }
}

```
